### PR TITLE
Ship v1.97.7: /change-job role transitions and the /goal-backlog planning layer

### PR DESCRIPTION
**Pull request by davekilleen**
## Linked Issue
- Linear issue: none — built from beta feedback (reporter's vault ideas idea-022 and idea-023), specs and founder rulings in `docs/superpowers/specs/2026-09-03-*.md`

## What Changed
- **Reset safety:** `--onboard` over a completed vault now merges instead of replacing — every setting not re-answered carries forward; the dry run enumerates each changing key old → new; disabled capability rooms stay disabled; `pillars.yaml` keeps `priority_limits`/keywords; the completion marker keeps its original date.
- **Transition linter:** finalize on a completed vault captures a hash-manifested transition capsule first, then `verify_transition` fails loudly on any setting lost or changed outside the re-answered keys; `restore_transition_capsule` puts the two config files back exactly. New `/dex-doctor` quick check watches the latest capsule.
- **`/change-job` skill:** guided role transition orchestrating reset-with-carry-forward, `reroute_people` (new Work MCP tool that re-sorts People/Internal↔External from recorded emails, dry-run-first, never overwrites or guesses), archive passes to `07-Archives/Role_Transitions/`, per-project carry/close/park, task re-pillaring, and a closing ledger with per-pass undo. Natural-language triggers added to CLAUDE.md.
- **Task–goal backlog layer:** goal/project/weekly-priority filters on `list_tasks`; new `get_goal_backlog` grouped reader with staleness; `Next up` per-goal ordering (`set_task_next_up`); goal progress/planning context now reflect the task pool; new `/goal-backlog` grooming skill; `/week-plan` reads the groomed pool instead of prose guessing; `/triage` gains a queue-jump line.
- **Defect batch:** blocked-task query reads the real `- [b]` status; `check_goal_alignment`'s orphan branch implemented; `started` round-trips as `- [/]`; Tasks.md section conflict canonicalized (P0–P3) with `create_task` filing by priority; quarter-goals seed made parseable plus a lenient provisional fallback for freeform lists; Technical Guide's fictional inline-tag format replaced with the real child-bullet format.
- **Release:** Lens catalogue reconciled (2 new skill entries, pins refreshed, preview restamped), changelog written to the public bar, `scripts/release.sh patch` cut `v1.97.7` (tag pushed after merge).

## Test Plan
- Unit/integration tests added or updated: 6 new test files / ~90 new tests (reset carry-forward parity, transition capsule + verify/restore + tamper refusal, doctor probe states, goal-backlog readers/ordering, reroute_people incl. dry-run byte-identity and collision skips, instruction-honesty pins for both new skills); Lens count pins moved deliberately.
- Negative/error-path tests added or updated: verification-failure classes (lost/unexpected), tampered capsule refusal, collision/ambiguous/pinned-page skips, provisional-goal link refusals, crashed-vault replace path.
- Commands run locally: `pytest core/tests core/mcp/tests` (4,284 passed; 15 failures verified byte-identical on pristine main — container-environment artifacts: root-user permission semantics and a partial clone), all 6 Lens suites green (128), `npm run test:scripts` 189/189, provision node tests 20/20, `check-doc-drift.sh`, `score_skill.py` clean for both new skills.

## Ralph Wiggum Loop
- [x] I implemented the change.
- [x] I self-reviewed for defects and edge cases.
- [x] I requested specialist review for risky areas (testing/infra/security when relevant).
- [x] I addressed review findings and re-ran checks.

## Quality Gates
- [x] I added/updated tests or documented why no tests are needed.
- [x] I added a regression test for bug fixes, or this PR is not a bug fix.
- [x] I validated failure modes / edge cases.
- [x] I updated docs or confirmed no docs impact.
- [x] CI checks for lint + tests + coverage are expected to pass.

## Risk & Rollback
- Risk level (low/medium/high): medium — the reset merge changes long-standing behavior, but only for vaults that completed onboarding, and the new capsule/verifier makes any regression visible and restorable; all other changes are additive tools, skills and readers.
- Rollback plan: revert the merge; the release tag `v1.97.7` is pushed only after this merge, so no public release exists until then. For users, `/dex-rollback` rewinds the adoption and `restore_transition_capsule` restores reset-touched configs.

## Docs Impact
- Files updated: CHANGELOG.md (public notes for 1.97.7), Dex_Technical_Guide.md (real task format + Next up), Folder_Structure.md, `.claude/skills/README.md`, CLAUDE.md (trigger block + catalog lines), two design specs + orchestration plan under `docs/superpowers/`.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01SoPX4HCJeAPTfss8shwVQs

---
_Generated by [Claude Code](https://claude.ai/code/session_01SoPX4HCJeAPTfss8shwVQs)_

---

> [!NOTE]
> **Medium Risk**
> Reset merge and guided file moves affect `user-profile.yaml`, pillars, people folders, and planning pages on completed vaults; transition capsules and preview-first skills reduce but do not remove the impact of mis-runs.
> 
> **Overview**
> **v1.97.7** adds a **goal-by-goal task backlog** and a **guided job-change flow**, and makes **reset/onboard safe** on vaults that already finished setup.
> 
> New **`/goal-backlog`** walks open tasks per quarter goal via `get_goal_backlog`: confirm tentative links, retire stale work (done, Someday move, or per-task delete with line preview), and set **`Next up`** order with `set_task_next_up`. **`/week-plan`** now pulls `next_up_tasks` from `get_weekly_planning_context` instead of inventing goal work; **`/triage`** optionally asks whether a new goal task should jump a groomed queue.
> 
> New **`/change-job`** orchestrates role moves: reset with carry-forward, `reroute_people` (new `core/entity_engine/reroute.py` + Work MCP, dry-run first), archive old goals/priorities/projects under `07-Archives/Role_Transitions/`, re-point task pillars, and close with an undo ledger. **`/reset`** is narrowed to same-job preference changes and documents **transition capsules** (`core/customization_migration/transition.py`): snapshot before rewrite, `verify_transition`, `restore_transition_capsule`.
> 
> Docs and seeds align with the real task model (caret IDs, child bullets for Goal/Pillar/**Next up**); quarter-goals starters show structured goal headings; CHANGELOG and lens catalog register both skills; analytics adds `goal_backlog_completed` and `change_job_completed`.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit e8327d2542e8aea007c0135ede1a1de007eeeab9. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Comment by davekilleen** [comment]
**Status: one deliberate red, waiting on a founder action — no code change pending.**

The only failing check on head `7298236` is the quality job's **Old-version release-tag reachability gate**. It refuses any new stable release right now by design: versions v1.62–v1.74 shipped an update checker that goes silent once **32** newer `dist/release/*` tags exist, the count stands at **31**, and publishing v1.97.7 would consume the final slot. The gate demands the verified bridge-and-archive procedure (see `docs/UPDATE-RESCUE.md`) on the public tag list first — a founder action, not something this PR should work around.

Everything else is green on this head: PII gate, path-contract, doc drift, architecture inventory, lint, distribution checks (package version matches the changelog heading), lens catalogue dry-run, and both portable-plugin builds; the test shards passed on the two prior heads whose only deltas since are a one-line lint fix and the release-artifact restore. This PR is not half-releasable by design — the catalogue requires new skills' first-release version as a shipped changelog heading, and distribution requires that heading to match the package version — so the v1.97.7 cut rides here and this gate is intentionally the last thing standing.

Once the bridge-and-archive is done, this head goes green with zero edits; then: merge (merge commit, so the release commit stays reachable), tag `v1.97.7`, and the draft-then-verified-publish pipeline takes it live.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: core/mcp/onboarding_server.py [hunk 1]
```diff
@@ -550,15 +550,44 @@ def check_granola() -> Dict[str, Any]:
...
+    # default applies only when neither an answer nor a recorded state exists.
+    current = _completed_profile_capability_states()
     states: Dict[str, bool] = {}
     for room in capability_rooms.room_ids():
         if room in selected:
             states[room] = selected[room] is True
             continue
+        if room in current:
+            states[room] = current[room]
+            continue
```
[the remark this answers is not in the document]
Comment (davekilleen) [reply core/mcp/onboarding_server.py:605]: Confirmed and fixed in e8327d2: `_completed_profile_capability_states` now mirrors the real read path — the explicit `capabilities.<room>.enabled` key first, then the room's legacy config switch (`quarterly_planning.enabled` for `quarter_goals`) — so a legacy-only disable counts as a recorded choice and survives a reset. Regression test added covering both the legacy disable and the new key winning over it.

---
_Generated by [Claude Code](https://claude.ai/code)_
